### PR TITLE
(INT): UnElideLifetimes for methods

### PR DESCRIPTION
**Pull request by ujpv**
Small fix for methods declarations and implementation. And tests.

### CODE REVIEW

File: src/test/kotlin/org/rust/ide/intentions/UnElideLifetimesIntentionTest.kt [hunk 1]
```diff
@@ -55,6 +55,43 @@ class UnElideLifetimesIntentionTest : RustIntentionTestBase(UnElideLifetimesInte
...
+    fun testMethodDecl() = doAvailableTest(
+        """
+        trait Foo {
+            fn /*caret*/bar(&self, x: &i32, y: &i32, x: i32) -> &i32;
+        }
+        """
+        ,
+        """
+        trait Foo {
+            fn /*caret*/bar<'a, 'b>(&self, x: &'a i32, y: &'b i32, x: i32) -> &'a i32;
```
Comment (matklad) [inline src/test/kotlin/org/rust/ide/intentions/UnElideLifetimesIntentionTest.kt]: It should be `&'a self`.

File: src/test/kotlin/org/rust/ide/intentions/UnElideLifetimesIntentionTest.kt [hunk 2]
```diff
@@ -55,6 +55,43 @@ class UnElideLifetimesIntentionTest : RustIntentionTestBase(UnElideLifetimesInte
...
+        }
+        """
+        ,
+        """
+        trait Foo {
+            fn bar(&self, x: &i32, y: &i32, x: i32) -> &i32;
+        }
+        struct S {}
+        impl Foo for s {
+            fn /*caret*/bar<'a, 'b>(&self, x: &'a i32, y: &'b i32, x: i32) -> &'a i32 {
```
Comment (matklad) [inline src/test/kotlin/org/rust/ide/intentions/UnElideLifetimesIntentionTest.kt]: ditto

File: src/test/kotlin/org/rust/ide/intentions/UnElideLifetimesIntentionTest.kt (hunk 1 above)
Comment (ujpv) [inline src/test/kotlin/org/rust/ide/intentions/UnElideLifetimesIntentionTest.kt]: Sure!
I made a mistake somewhere...
Thanks.

**Comment by ujpv** [comment]
Now &self-agr handled correctly.

File: src/test/kotlin/org/rust/ide/intentions/UnElideLifetimesIntentionTest.kt [hunk 3]
```diff
@@ -55,6 +55,43 @@ class UnElideLifetimesIntentionTest : RustIntentionTestBase(UnElideLifetimesInte
...
+            }
+        }
+        """
+        ,
+        """
+        trait Foo {
+            fn bar(&self, x: &i32, y: &i32, x: i32) -> &i32;
+        }
+        struct S {}
+        impl Foo for s {
```
Comment (matklad) [inline src/test/kotlin/org/rust/ide/intentions/UnElideLifetimesIntentionTest.kt]: s/s/S

Comment (ujpv) [inline src/test/kotlin/org/rust/ide/intentions/UnElideLifetimesIntentionTest.kt]: Ok :)

**Comment by matklad** [comment]
:+1:

**Pull request merged**